echo/test/unit/echo_client_test.js

```diff
@@ -90,123 +90,35 @@ describe("Echo Node Client Test Suite", function(){
-            var echoClient = echo.createClient({
-                echo_endpoint: endPoint
-            });
-
-            var requestStub = sandbox.stub(request, 'post')
-            requestStub.callsFake(function (options, callback) {
-                options.headers.should.have.property('User-Agent', 'talis-node/0.2.1');
-                if(options.method !== 'POST'){
-                    var error = new Error('Invalid field: options.method');
-                    callback(error);
-                } else{
-                    callback(null);
-                }
-            });
-
-            echoClient.addEvents('secret', {class:'class', source:'source'}, function(err){
-                (err === null).should.be.false;
-                done();
-            });
-        });
-        it("- add events should return an error if call to request has missing option.json", function(done){
-            var echoClient = echo.createClient({
-                echo_endpoint: endPoint
-            });
-
-            var requestStub = sandbox.stub(request, 'post')
-            requestStub.callsFake(function (options, callback) {
-                options.headers.should.have.property('User-Agent', 'talis-node/0.2.1');
-                if(!options.json){
-                    var error = new Error('Missing field: options.json');
-                    callback(error);
-                } else{
-                    callback(null);
-                }
-            });
-
-            echoClient.addEvents('secret', {class:'class', source:'source'}, function(err){
-                (err === null).should.be.false;
-                done();
-            });
-        });
         it(" - should throw an error if persona validation fails", function(done) {
             var echoClient = echo.createClient({
                 echo_endpoint: endPoint
             });
 
-            var requestStub = sandbox.stub(request, 'post')
-            requestStub.callsFake(function (options, callback) {
-                options.headers.should.have.property('User-Agent', 'talis-node/0.2.1');
-                callback(
-                  null,
-                  {statusCode: 401},
-                  {
-                      "error": "invalid_token"
-                  }
-              );
-            });
+            nock(endPoint)
+                .post('/1/events')
+                .reply(function(uri, requestBody){
+                    this.req.headers['user-agent'].should.equal('talis-node/0.2.3');
+                    return [401, {"error": "invalid_token"}];
+                });
 
             echoClient.addEvents("incorrect_token", {class:'class', source:'source'}, function(err, result) {
                 (err === null).should.be.false;
 
-                err.code.should.equal(8);
-                err.label.should.equal("INVALID_TOKEN");
+                err.code.should.equal(6);
+                err.label.should.equal("UNAUTHORISED");
```

Comment:
8 / INVALID_TOKEN -> 6 / UNAUTHORIZED

The mocking is returning the same 401 response. Is this something caused by the change of http client library?

This isn't talking about AUTHENTICATION/AUTHORISATION - so what is the difference between INVALID_TOKEN and UNAUTHORISED?